modules/admin/con-rhdh-operator-deployment-config.adoc

```diff
@@ -0,0 +1,56 @@
+// Module included in the following assembly:
+// assembly-rhdh-operator-deployment-config
+
+[id="con-rhdh-operator-deployment-config_{context}"]
+= {product} Operator deployment configuration
+
+Since v1alpha2 (Operator version 0.3.0), the Backstage Custom Resource Definition (CRD) contains `spec.deployment`. This allows for patching the Backstage Deployment resource with fields defined in `spec.deployment.patch`, which contains a fragment of the `apps.Deployment` object.
+
+This patching is performed by strategic merge patch using Kustomize's library.
+
+For example, the following specification fragment will:
+
+* Set an additional volume named `my-volume` and mount it to `/my/path` of the Backstage container.
+* Replace existed `dynamic-plugins-root` volume.
+* Set the CPU request for Backstage containers to 250m.
+* Set the label `my=true` for the Backstage Pod.
+* Adds another `my-sidecar` container.
+
+[NOTE]
+====
+It is assumed that the Backstage container name as defined in the default configuration is `backstage-backend`.
```

Comment:
```suggestion
To learn more about merging, see [Strategic Merge Patch](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md#basic-patch-format).
```

`backstage-backend` is the name of the RHDH container in the operator default configuration, and it won't change in most cases. I am not sure this needs to be stated here. Instead, I am proposing a note to learn more about how the merging is done, as this can provide additional context and examples.